sdk/azidentity/aad_identity_client.go

```diff
@@ -45,21 +45,20 @@ type interactiveConfig struct {
 // aadIdentityClient provides the base for authenticating with Client Secret Credentials, Client Certificate Credentials
 // and Environment Credentials. This type includes an azcore.Pipeline and TokenCredentialOptions.
 type aadIdentityClient struct {
-	options  TokenCredentialOptions
-	pipeline azcore.Pipeline
+	authorityHost string
+	pipeline      azcore.Pipeline
 }
 
 // newAADIdentityClient creates a new instance of the aadIdentityClient with the TokenCredentialOptions
 // that are passed into it along with a default pipeline.
 // options: TokenCredentialOptions that can configure policies for the pipeline and the authority host that
 // will be used to retrieve tokens and authenticate
-func newAADIdentityClient(options *TokenCredentialOptions) (*aadIdentityClient, error) {
+func newAADIdentityClient(authorityHost string, options *pipelineOptions) (*aadIdentityClient, error) {
```

Comment:
Since this isn't public surface area, you could opt to made `options` by-value as we have complete control over the input.  Not sure how useful the change would be but it does appear to simplify the `nil` case below (i.e. that goes away).

Reply:
Done